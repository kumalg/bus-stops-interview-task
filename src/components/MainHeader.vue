<template>
    <header class="main-header">
        <h1 class="main-header-title">{{ title }}</h1>
    </header>
</template>

<script setup lang="ts">
defineProps<{
    title: string;
}>();
</script>

<style lang="scss" scoped>
.main-header {
    padding: 0.25rem 0;

    .main-header-title {
        font-weight: 600;
        font-size: 1.5rem;
        line-height: 2rem;
        margin: 0;
    }
}
</style>
