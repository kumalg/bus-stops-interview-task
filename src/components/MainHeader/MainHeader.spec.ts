import { shallowMount } from '@vue/test-utils';

import MainHeader from './MainHeader.vue';

const EXAMPLE_TITLE = 'EXAMPLE_TITLE';

describe('MainHeader', () => {
    const MainHeaderFactory = (props: InstanceType<typeof MainHeader>['$props']) => {
        return shallowMount(MainHeader, {
            props
        });
    };

    it('renders proper html markup', () => {
        const wrapper = MainHeaderFactory({ title: EXAMPLE_TITLE });

        expect(wrapper.element).toMatchSnapshot();
    });

    it('shows proper title', () => {
        const wrapper = MainHeaderFactory({ title: EXAMPLE_TITLE });

        expect(wrapper.find('.main-header-title').text().trim()).toBe(EXAMPLE_TITLE);
    });
});
