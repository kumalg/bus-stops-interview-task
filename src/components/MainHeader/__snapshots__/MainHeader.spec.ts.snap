// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHeader renders proper html markup 1`] = `
<header
  class="main-header"
>
  <h1
    class="main-header-title"
  >
    EXAMPLE_TITLE
  </h1>
</header>
`;
